pkg/utils/test/assertions.go

```diff
@@ -69,7 +69,15 @@ func assertOption(fns ...AssertOptionFn) AssertOption {
 	return option
 }
 
-func (f Framework) AssertResourceExits(name, ns string, obj client.Object, fns ...AssertOptionFn) {
+func (f Framework) WaitUntil(msg string, fn wait.ConditionFunc, fns ...AssertOptionFn) {
```

Comment:
rename this function to hint that there is an assert inside?

Reply:
Any suggestion?

Reply:
@vimalk78 , I am going to merge this as is, we can rename this test utility when we have a better idea :)

Reply:
I missed seeing the the function signature, which has `AssertOptionFn` so there should be no ambiguity.